### DISCUSSION TITLE
THANK YOU!!!!

### DISCUSSION THREAD

**Discussion by smonska** [yermak/AudioBookConverter#33] [General]
This software saved me hours of time in converting my library from mp3 to m4b. I would not have done it except for all the hard work you put into this.  Thank you! 

A few suggestions for future builds if I might ask...
1. While in the "Book Info" tab selecting the "Enter" key would "Start" the process automatically. You would not have to click on the Start button that way.  
2. If you could create a menu/options selection where you could list a default "Save To" folder.  This way you would not have to select the same folder each time you select "Start".
3. When selecting the Add "File/Folder" option Windows Explorer pops up. You then have to click on the file or folder you want to import and then select the "Select Folder" button .  It would be quicker if it allowed for "double clicking" on the file/folder you wanted to import to automatically "Select Folder" for you and import your highlighted selection. Bypassing the need to click on the folder or  file and then clicking on the "Select Folder" button.  This would only work when selecting a Folder or a single File. When selecting multiple Files to import you would still have to highlight them and click on the "Select Folder" button.  

Total pie in the sky request...
Not sure how this would be done but... Allow for imports by directory.  Each directory would be Queued separately and converted into a single file. One file converted for each directory. Might work like this... If you had a third option on the Add button called Folder AND Subdirectories, this would import all files in the parent and it's subfolders.  Then on the Conversion Mode tab you add a new selection "Convert each folder into a separate (single) file." I know it's not as easy to do on the back end but it would save time while importing books series where every edition is in a separate folder.  

Just a few suggestions to speed up the import process for large libraries. Anyway thanks again for the great software,
Steve

**Comment by smonska** [comment]
Oh one last thing and this is TOTAL pie in the sky but...
Allow for metadata tag mapping.

**Comment by yermak** [comment]
Steve, thank you very much for warm words.

As I understand you demand as comming from desire to convert large library at once...

1) It's probably possible, but during improvement of the program interface will become more heavier (unfortunately) - see version 4 with chapters support, so this particular extra click should not make big difference.
2) Well, it will make sense if I make name selection completely autmatic - then we can skip both steps selecting output directory and correcting file name. As for now i believe default behaviour is output directory is the last used output directory, so if you used it once - next time it will suggest the same directory. Let me know if does not work in your case
3) File and folder selection dialog as system - rewriting them manually does not bring much value. Also not sure i understand your idea of double clicking to folder to select it - if you do this you will just go inside it.
4) As for this request - version 4 brings some improvements, not fully result your problem, but if you have series of the books in the one folder - you can import it (including all nested directories) and then with few clicks split them into parts, which will be queued for conversion separately.
5) FAs for metadata tag mapping - do you mean config file to map mp3 tags into m4b book attributes. I followed iTunes concept of mapping, however, majority of mp3 files use tags quite randomly - thus i stick with most correct version. In general this is doable, but not sure if this will be widely used.

**Comment by yermak** [comment]
This topic is closed - discussions and feedback are moved here https://github.com/yermak/AudioBookConverter/discussions
